internal/resources/metal/organization_member/models.go

```diff
@@ -0,0 +1,86 @@
+package organizationmember
+
+import (
+	"context"
+	"fmt"
+	"path"
+
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+
+	"github.com/packethost/packngo"
+)
+
+type ResourceModel struct {
+	ID             types.String `tfsdk:"id"`
+	Invitee        types.String `tfsdk:"invitee"`
+	InvitedBy      types.String `tfsdk:"invited_by"`
+	OrganizationID types.String `tfsdk:"organization_id"`
+	ProjectsIDs    types.Set    `tfsdk:"projects_ids"`
+	Nonce          types.String `tfsdk:"nonce"`
+	Message        types.String `tfsdk:"message"`
+	Created        types.String `tfsdk:"created"`
+	Updated        types.String `tfsdk:"updated"`
+	Roles          types.Set    `tfsdk:"roles"`
+	State          types.String `tfsdk:"state"`
+}
+
+func (m *ResourceModel) parse(ctx context.Context, invitee string, members []packngo.Member, invitations []packngo.Invitation) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	member, err := FindMember(invitee, members, invitations)
+	if err != nil {
+		m.OrganizationID = basetypes.NewStringNull()
+		return diags
+	}
+
+	if member.isMember() {
+		projectsList, _ := types.SetValueFrom(ctx, types.StringType, member.Member.Projects)
+		m.ProjectsIDs = projectsList
+		m.State = types.StringValue("active")
+
+		rolesList, _ := types.SetValueFrom(ctx, types.StringType, member.Member.Roles)
+
+		m.Roles = rolesList
+		m.OrganizationID = types.StringValue(member.Member.Organization.URL)
+
+	} else if member.isInvitation() {
+
+		if len(member.Invitation.Projects) != 0 {
```

Comment:
We probably don't want this length check.  If projects is an empty array we should end up with `m.ProjectsIDs` as an empty array; with this length check, we'll end up with nil for `m.ProjectsIDs` which could be the source of "invalid value after apply" errors.